filebeat/input/file/identifier.go

```diff
@@ -0,0 +1,127 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package file
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strconv"
+	"strings"
+
+	"github.com/mitchellh/hashstructure"
+
+	"github.com/elastic/beats/v7/libbeat/logp"
+)
+
+type StateIdentifier interface {
+	// ID generates and returns the ID of the state
+	ID(*State) string
+	// Identifier returns data to be persisted in the registry.
+	DataToSerialize() interface{}
```

Comment:
Why do we want to serialize additional data?

Reply:
It was needed for the ID generation. As we persist the ID now, it is unnecessary.